pkg/kapis/cluster/v1alpha1/handler.go

```diff
@@ -372,60 +381,42 @@ func (h *handler) validateKubeConfig(kubeconfig []byte) error {
 		return err
 	}
 
-	// clusters with the exactly same KubernetesAPIEndpoint considered to be one
+	// clusters with the exactly same kube-system namespace UID considered to be one
 	// MUST not import the same cluster twice
-	for _, cluster := range clusters {
-		if len(cluster.Spec.Connection.KubernetesAPIEndpoint) != 0 && cluster.Spec.Connection.KubernetesAPIEndpoint == config.Host {
-			return fmt.Errorf("existing cluster %s with the exacty same server address, MUST not import the same cluster twice", cluster.Name)
+	for _, existedCluster := range clusters {
```

Comment:
Could we add the clusterid to the status of Cluster object? I think we can use it in other scenarios too, it would also be easier to check clusterid here.

Reply:
I'm cool with this idea, but I think we should add it into an annotation, it's not a status message.

Reply:
Status usually set by controller, means watched state, which is exactly our case, I think it's fine to set it into status.

Reply:
I added a new `UID` field to status, PTAL

Reply:
Seen it, but there are too many changes on CRD yamls which shouldn't be. It's better to make changes only related to this RP.

Reply:
I have submitted a PR to update the automatically generated files separately, we need to merge that first: #4669

Reply:
Updated, PATL